app/assets/javascripts/Components/Result/autosave_text_form.jsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import { render } from "react-dom";
+
+function debounce(func, wait, immediate) {
+  var timeout;
+
+  return function executedFunction() {
+    var context = this;
+    var args = arguments;
+
+    var later = function () {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
+    };
+
+    var callNow = immediate && !timeout;
+
+    clearTimeout(timeout);
+
+    timeout = setTimeout(later, wait);
+
+    if (callNow) func.apply(context, args);
+  };
+}
+
+const SaveMessage = ({ unSaved }) => (
+  <div className="autosave-text">
+    {unSaved ? (
+      <p> There are unsaved changes </p>
```

Comment:
Please internationalize these strings (see https://guides.rubyonrails.org/i18n.html). You can put the strings under the `views/results/` locale file. It's okay to use English only.